### PR TITLE
x11-terms/kitty-terminfo: split out kitty-terminfo

### PR DESCRIPTION
**Pull request by end222**
Split out kitty-terminfo into a separate package as requested here: [https://bugs.gentoo.org/678486](https://bugs.gentoo.org/678486)

This is done to follow the recommendations given by the author of kitty about packaging here: [https://sw.kovidgoyal.net/kitty/build.html#note-for-linux-macos-packagers](https://sw.kovidgoyal.net/kitty/build.html#note-for-linux-macos-packagers)

Signed-off-by: Pablo Orduna <pabloorduna98@gmail.com>

### CODE REVIEW

**Comment by end222** [comment]
The upstream test functionality could be implemented, but it should be implemented in the kitty package and not in the kitty-terminfo package. This package would only be used to include the kitty terminfo file needed in servers where the clients connect using kitty.

The modifications done to the setup.py file make it possible to install only the file without having to compile the whole package but, as a result, the tests included for the whole package will fail because the compiled package is not really there.

It should be also working well with kitty-9999 but it may need some little modifications in the patch because setup.py is constantly changing.

**Comment by juippis** [comment]
You'll need to restrict tests with a comment above stating why, like
```
# kitty-terminfo is a split package from kitty so not all needed parts are built
RESTRICT="test"
```
You can word it better but you get the idea.

**Comment by end222** [comment]
Okay, understood!
I will change it as soon as possible.
Thank you very much!

File: x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit python-single-r1 toolchain-funcs xdg
+
+SRC_URI="https://github.com/kovidgoyal/kitty/releases/download/v${PV}/kitty-${PV}.tar.xz"
+S="${WORKDIR}/kitty-${PV}"
+
+KEYWORDS="~amd64 ~x86"
```
Comment (charIes17) [inline x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild]: Move KEYWORDS down between SLOT and IUSE

File: x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+DESCRIPTION="Terminfo for kitty, an OpenGL-based terminal emulator"
+HOMEPAGE="https://github.com/kovidgoyal/kitty"
```
Comment (charIes17) [inline x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild]: Move DESCRIPTION and HOMEPAGE before SRC_URI

Comment (end222) [reply x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild]: Okay, understood!

**Comment by juippis** [comment]
https://sw.kovidgoyal.net/kitty/build.html#note-for-linux-macos-packagers page says

>The terminfo file will be installed into linux-package/share/terminfo

But 
```
>>> Installing (1 of 1) x11-terms/kitty-terminfo-0.16.0::testworld
 * checking 2 files for package collisions
>>> Merging x11-terms/kitty-terminfo-0.16.0 to /
--- /usr/
--- /usr/lib64/
>>> /usr/lib64/kitty/
>>> /usr/lib64/kitty/terminfo/
>>> /usr/lib64/kitty/terminfo/x/
>>> /usr/lib64/kitty/terminfo/x/xterm-kitty
--- /usr/share/
--- /usr/share/terminfo/
--- /usr/share/terminfo/x/
>>> /usr/share/terminfo/x/xterm-kitty
>>> x11-terms/kitty-terminfo-0.16.0 merged.
```

So will it work? 

What if you have a machine that you use as a desktop AND a server? This now prevents installing kitty into it. 
So I'd like a 2nd commit updating kitty ebuild NOT to install this particular file, while revbumping it, in this PR.

File: x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="debug"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+RDEPEND="${PYTHON_DEPS}"
+
+DEPEND="${RDEPEND}"
+
+BDEPEND="virtual/pkgconfig"
```
Comment (juippis) [inline x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild]: This didn't seem to install any pkgconfig files, is this dep really needed, then?

File: x11-terms/kitty/kitty-0.16.0-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	insinto /usr
+	doins -r linux-package/*
+	dobin linux-package/bin/kitty
+	python_fix_shebang "${ED}"
+}
+
+pkg_postinst() {
+	xdg_icon_cache_update
+	optfeature "Displaying images in the terminal" virtual/imagemagick-tools
+}
```
Comment (juippis) [inline x11-terms/kitty/kitty-0.16.0-r2.ebuild]: You could add an `optfeature` for *kitty-terminfo* so it gains noticeability.

File: x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild (hunk 3 above)
Comment (end222) [reply x11-terms/kitty-terminfo/kitty-terminfo-0.16.0.ebuild]: That's right, I've done some test and it's not being used

**Pull request closed without merging**